### PR TITLE
chore: add typescript definition

### PR DESCRIPTION
**Pull request by overflowz**
This PR adds minimal types for the 5.x version. Feel free to update/extend.

N.B: Not well tested. If someone has time to test & approve, would be much appreciated.

### CODE REVIEW

File: lib/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+declare interface Socket extends EventEmitter, SocketAccessors {
+  type: keyof SocketType;
+  pause: () => void;
+  resume: () => void;
+  ref: () => void;
+  unref: () => void;
+  read: () => null | Buffer[];
+  setsockopt: (opt: keyof SocketOptions, value: unknown) => this;
+  getsockopt: (opt: keyof SocketOptions) => unknown;
+  readonly closed: boolean,
```
Comment (overflowz) [inline lib/index.d.ts:87]: typo, should be `;` instead.

**Review by aminya — CHANGES_REQUESTED** [review]
Could you instead edit the v4 types instead of writing this from scratch? The types here lack documentation and test.

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/zeromq/index.d.ts

**Comment by overflowz** [comment]
@aminya Hi, shall I create a pull request directly to DefinitelyTyped or include it here? (asking because I never did anything like that before).

**Comment by aminya** [comment]
> @aminya Hi, shall I create a pull request directly to DefinitelyTyped or include it here? (asking because I never did anything like that before).

It seems easier to make a PR to DefinitelyTyped because you can just edit the types instead of starting from scratch. They also have tests. For v6, we will deprecate `@types/zeromq` as we are already using TypeScript.

**Comment by overflowz** [comment]
Cool, so we can close this PR without merging and I'll just edit the existing one (v4 on DefinitelyTyped).

Feel free to close it.

**Comment by aminya** [comment]
Thank you for doing this!

**Pull request closed without merging**